### PR TITLE
[Feature] Be able to show quick code samples for getting started

### PR DESCRIPTION
**Pull request by pksjce**
The examples here have the full working code in [https://github.com/pksjce/webpackdocs-getting-started](https://github.com/pksjce/webpackdocs-getting-started)
The result looks like below.
You can change the common scenarios using the dropdown and I show different code snippets for them.
screen shot 2016-10-07 at 6 04 34 pm

### CODE REVIEW

File: components/ShowCode.jsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import React, {Component} from 'react';
+import Snippet from './Snippet.jsx';
+
+export default class ShowCode extends Component {
+    constructor(){
+        super();
+        this.options = [{
```
Comment (pksjce) [inline components/ShowCode.jsx]: @bebraw - Need to push these templates to some other place ideally

File: styles/components/_showcode.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
+.showCode {
```
Comment (pksjce) [inline styles/components/_showcode.scss]: @bebraw - I want to use styles from here. But I'm not sure how its connected to the jsx file.

File: components/Snippet.jsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import Highlight from 'react-highlight';
+require('style!css!highlight.js/styles/github-gist.css');
```
Comment (pksjce) [inline components/Snippet.jsx]: @bebraw - Is there a better way to do this?

File: styles/components/_showcode.scss (hunk 2 above)
Comment (bebraw) [inline styles/components/_showcode.scss]: Relying on class names should be enough. I know it's global CSS but it's not too bad. 🎱

File: components/Snippet.jsx (hunk 3 above)
Comment (bebraw) [inline components/Snippet.jsx]: Here, https://github.com/webpack/webpack.js.org/blob/develop/webpack.config.js#L9 . You can point from the entry.

**Comment by gregvenech** [comment]
Looks like a great start to me. Maybe we should show a css or json import because I feel like webpack allowing the modularization of assets as well is one of the strongest points for using it over other tools. That said we'd still have to find a way to keep it simple. I'll try to pull your branch and play around with it a bit more today.

**Comment by jhnns** [comment]
We could use [webpackbin](http://www.webpackbin.com/) by @christianalfoni for that

**Comment by pksjce** [comment]
Webpackbin looks really good. Where can I see the generated webpack config file?

**Comment by pksjce** [comment]
@gregvenech - Did you get any time to evaluate this idea? I have three code examples as of now
1. Simple setup [entry/output]
2. Setup with babel [entry/output/babel-loader]
3. Setup with css [entry/styles/css-loader]

File: components/ShowCode.jsx (hunk 1 above)
Comment (pksjce) [inline components/ShowCode.jsx]: Made a separate snippets folder and passed it through highlight-loader!

File: components/Snippet.jsx (hunk 3 above)
Comment (pksjce) [inline components/Snippet.jsx]: This worked!

**Comment by gregvenech** [comment]
@pksjce sorry the last couple days have been hectic for me. Now that I'm finished with #225 I'm going to start working on the visualization section of the splash page. I'll pull your branch and test it out as well. Mind if I PM you on gitter so we can discuss this page more and try to get something up and running?

I came across webpackbin recently as well. If they have an embed option maybe thats an easier fix than doing everything custom. Either way I think we should link to webpackbin throughout the docs as it would definitely be a great resource for learning to use webpack.

**Comment by bebraw** [comment]
@christianalfoni Any thoughts on webpackbin embeds to webpack docs?

**Comment by pksjce** [comment]
@gregvenech - Sure. I pinged you on gitter myself :)
I looked at webpackbin but I didnt see a way to actually see the `webpack.config.js` that is generated. Maybe I missed something.

**Comment by pksjce** [comment]
Decided to close this for now. We will use webpackbin and try to enhance that to our needs.

**Comment by bebraw** [comment]
Having something simple and standalone has still value. The problem is that [webpackbin hasn't seen active development in months](https://github.com/christianalfoni/webpack-bin) and the embed problem is still waiting for an answer.

**Comment by pksjce** [comment]
People copy/paste webpack config all the time. It would be awesome if they could copy paste it right out of the docs.

**Comment by pksjce** [comment]
@skipjack - Maybe we could revisit this PR now that #264 is closed?

**Comment by skipjack** [comment]
Definitely, from my memory of testing it out a while ago I think if we just tweak the example code a bit and update the design it could work well. I think the important thing is to keep it fairly simple so it doesn't become a heavy burden to maintain, but still functional enough that it's useful for beginners to play around with.

Do you have an idea where this would go in the current version of the site (e.g. replace the current splash page example, or somewhere in getting-started maybe)?

**Pull request closed without merging**